### PR TITLE
Added content_test.go to increase test coverage.

### PR DESCRIPTION
**Pull request by Skarlso**
Proper pull request. For all the conversation please see pull request #1812.

### CODE REVIEW

File: helpers/content.go [hunk 1]
```diff
@@ -35,10 +35,10 @@ import (
 	"sync"
 )
 
-// Length of the summary that Hugo extracts from a content.
+// SummaryLength of the summary that Hugo extracts from a content.
```
Comment (moorereason) [inline helpers/content.go]: s/of/is the length of/

File: helpers/content.go [hunk 2]
```diff
@@ -35,10 +35,10 @@ import (
...
 var SummaryLength = 70
 
-// Custom divider <!--more--> let's user define where summarization ends.
+// SummaryDivider Custom divider <!--more--> let's user define where summarization ends.
```
Comment (moorereason) [inline helpers/content.go]: SummaryDivider denotes where content summarization should end. The default is "".

File: helpers/content.go [hunk 3]
```diff
@@ -157,7 +157,7 @@ func BytesToHTML(b []byte) template.HTML {
 	return template.HTML(string(b))
 }
 
-// GetHtmlRenderer creates a new Renderer with the given configuration.
+//GetHTMLRenderer creates a new Renderer with the given configuration.
```
Comment (moorereason) [inline helpers/content.go]: `s/Get/ Get/`

File: helpers/content.go [hunk 4]
```diff
@@ -237,7 +237,7 @@ func markdownRenderWithTOC(ctx *RenderingContext) []byte {
 		getMarkdownExtensions(ctx))
 }
 
-// mmark
+//GetMmarkHtmlRenderer returns markdown html renderer.
```
Comment (moorereason) [inline helpers/content.go]: `s/Get/ Get/`

File: helpers/content.go [hunk 5]
```diff
@@ -259,6 +259,7 @@ func GetMmarkHtmlRenderer(defaultFlags int, ctx *RenderingContext) mmark.Rendere
 	}
 }
 
+//GetMmarkExtensions returns markdown extensions.
```
Comment (moorereason) [inline helpers/content.go]: `s/Get/ Get/`

File: helpers/content.go [hunk 6]
```diff
@@ -283,17 +284,12 @@ func GetMmarkExtensions(ctx *RenderingContext) int {
 	return flags
 }
 
+//MmarkRender renders markdowns.
```
Comment (moorereason) [inline helpers/content.go]: `s/Mmark/ Mmark/`

File: helpers/content.go [hunk 7]
```diff
@@ -331,7 +327,7 @@ func ExtractTOC(content []byte) (newcontent []byte, toc []byte) {
 }
 
 // RenderingContext holds contextual information, like content and configuration,
-// for a given content renderin.g
+// for a given content renderin.
```
Comment (moorereason) [inline helpers/content.go]: `s/renderin./rendering./`

File: helpers/content_test.go [hunk 8]
```diff
@@ -117,3 +122,296 @@ func TestTruncateWordsByRune(t *testing.T) {
...
+
+	if bytes.Equal(headerBuffer.Bytes(), expectedHeaderID) {
+		t.Errorf("Header Id Postfix applied. Actual:%s Expected:%s", headerBuffer.String(), expectedHeaderID)
+	}
+}
+
+func TestGetMarkdownExtensionsMasksAreRemovedFromExtensions(t *testing.T) {
+	type data struct {
+		testFlag int
+	}
```
Comment (moorereason) [inline helpers/content_test.go]: This `data` type looks unused.  Delete.

**Comment by bep** [comment]
As to commit message, see https://github.com/spf13/hugo/blob/master/CONTRIBUTING.md

**Comment by Skarlso** [comment]
Commit fixed. I didn't fix the commits as a whole. Now I got it. Thanks!

**Comment by bep** [comment]
Yes, now it looks good (I modified your commit message slightly). Thanks for your contribution. Tests are very, very important.

Merged in e95f3af

**Comment by Skarlso** [comment]
Ahh thanks, I learned a lot from this. I'm working on the next batch already. I love writing tests actually. 

Cheers @bep!

**Pull request closed without merging**